.helm/ecamp3/templates/api_cache_deployment.yaml

```diff
@@ -0,0 +1,116 @@
+        {{- include "apiCache.selectorLabels" . | nindent 8 }}
+      annotations:
+        checksum/vclConfigmap: {{ include (print $.Template.BasePath "/api_cache_vcl_configmap.yaml") . | sha256sum }}
+        rollme: {{ .Values.imageTag | quote }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "app.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      enableServiceLinks: false
+      containers:
+        - name: {{ .Chart.Name }}-api-cache-varnishd
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.apiCache.image.repository }}:{{ .Values.apiCache.image.tag | default .Values.imageTag }}"
+          imagePullPolicy: {{ .Values.apiCache.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.apiCache.varnishHttpPort }}
+              protocol: TCP
+            - name: purge
+              containerPort: {{ .Values.apiCache.varnishPurgePort }}
+              protocol: TCP
+          env:
+            - name: VARNISH_SIZE
+              value: "{{ .Values.apiCache.varnishSize }}"
+            - name: VARNISH_HTTP_PORT
+              value: "{{ .Values.apiCache.varnishHttpPort }}"
+            - name: COOKIE_PREFIX
+              value: {{ include "api.cookiePrefix" . | quote }}
+          args:
+            - -a
+            - {{ printf ":%d,HTTP" (.Values.apiCache.varnishPurgePort | int) }}
+            - -p
+            - http_max_hdr=96
+          resources:
+            {{- toYaml .Values.apiCache.resources | nindent 12 }}
+          volumeMounts:
+            - name: vcl-configmap
+              mountPath: /etc/varnish
+            - name: vsm
+              mountPath: /var/lib/varnish
+        {{- if .Values.apiCache.logging.enabled }}
+        - name: {{ .Chart.Name }}-api-cache-varnishncsa
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.apiCache.image.repository }}:{{ .Values.apiCache.image.tag | default .Values.imageTag }}"
+          imagePullPolicy: {{ .Values.apiCache.image.pullPolicy }}
+          command:
+          - varnishncsa
+          - -b
+          - -c
+          {{- if .Values.apiCache.logging.customOutputJsonFormat }}
+          - -j
+          {{- end }}
+          {{- if .Values.apiCache.logging.customOutput }}
+          - -F
+          - {{ .Values.apiCache.logging.customOutput | squote }}
+          {{- end }}
+          - -t
+          - {{ .Values.apiCache.logging.timeout | quote }}
+          resources:
+            {{- toYaml .Values.apiCache.logging.resources | nindent 12 }}
+          volumeMounts:
+          - name: vsm
+            mountPath: /var/lib/varnish
+        {{- end }}
+        {{- if .Values.apiCache.prometheus.enabled }}
```

Comment:
Yep, it's 2 different systems and 2 different purposes.

**Logging**: Outputs a stream of access logs (1 line for each client request; 1 line for each backend request, if needed). This access log provides information on individual requests. This log should be collected by BacLuc's EFFK stack and could be used for error debugging, but also to derive some statistics (e.g. how many miss vs. hit on a specific endpoint)

**Prometheus exporter**: Provides various statistics and status parameters of Varnish server. This can be read by Prometheus/Grafana and provides some insights into the Varnish server in general, e.g.
- overall number of hits
- overall number of misses
- number of objects in the cache
- used memory
- etc.
All this data from prometheus exporter is aggregated data, so there's no information on specific endpoints/routes.